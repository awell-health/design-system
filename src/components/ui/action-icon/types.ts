export enum ActionType {
  ApiCall = 'API_CALL',
  ApiCallGraphql = 'API_CALL_GRAPHQL',
  Calculation = 'CALCULATION',
  Checklist = 'CHECKLIST',
  ClinicalNote = 'CLINICAL_NOTE',
  Form = 'FORM',
  Message = 'MESSAGE',
  Plugin = 'PLUGIN',
  PushToEmr = 'PUSH_TO_EMR'
}
