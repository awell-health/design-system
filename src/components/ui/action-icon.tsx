import * as React from "react"
import { cn } from "@/lib/utils"
import { Icon, IconType } from "./icon"


export enum ActionType {
  ApiCall = "API_CALL",
  ApiCallGraphql = "API_CALL_GRAPHQL",
  Calculation = "CALCULATION",
  Checklist = "CHECKLIST",
  ClinicalNote = "CLINICAL_NOTE",
  Form = "FORM",
  Message = "MESSAGE",
  Plugin = "PLUGIN",
  PushToEmr = "PUSH_TO_EMR",
}

export interface ActionIconProps {
    actionType?: ActionType
    icon_url?: string
    size?: "sm" | "lg"
}

const ActionIcon = React.forwardRef<HTMLDivElement, ActionIconProps>(
  ({ ...props }, _ref) => {
    const TYPES_ICONS_MAP: Record<
      ActionType,
      IconType
    > = {
      [ActionType.ApiCall]: "RiTerminalBoxFill",
      [ActionType.ApiCallGraphql]: "RiTerminalWindowFill",
      [ActionType.Calculation]: "RiCalculatorFill",
      [ActionType.Checklist]: "RiCheckboxMultipleFill",
      [ActionType.ClinicalNote]:  "RiHealthBookFill",
      [ActionType.Form]: "RiSurveyFill",
      [ActionType.Message]: "RiMessage3Fill",
      [ActionType.Plugin]: "RiUploadCloudFill",
      [ActionType.PushToEmr]: "RiUploadCloudFill"
    }

    const { icon_url, actionType, size='sm' } = props

    const icon =
      TYPES_ICONS_MAP[actionType as keyof typeof TYPES_ICONS_MAP] ??
      "RiQuestionFill"

    const color = icon === "RiQuestionFill" ? "fill-slate-400" : "fill-slate-800";
    const iconSize = size === 'sm' ? 'w-4 h-4' : 'w-6 h-6'
    const containersize = size === 'sm' ? 'w-6 h-6' : 'w-9 h-9'

    if (actionType === ActionType.Plugin && icon_url) {
      return (
        <img
          className={`${containersize} rounded-md border border-slate-200`}
          src={icon_url}
        />
      )
    }

    return (
      <div className={`${containersize} flex bg-slate-100 rounded-md border border-slate-200 items-center justify-center`}>
        <Icon icon={icon} className={cn(`${iconSize}`, color)} />
      </div>
    )
  },
)
ActionIcon.displayName = "ActionIcon"

export { ActionIcon }
