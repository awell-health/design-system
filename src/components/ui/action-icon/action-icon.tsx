import * as React from 'react';
import { cn } from '@/lib/utils';
import { Icon, IconType } from '../icon';
import { ActionType } from './types';

interface ActionIconProps {
  actionType?: ActionType;
  icon_url?: string;
  size?: 'sm' | 'lg' | 'xl';
}

const ActionIcon = React.forwardRef<HTMLDivElement, ActionIconProps>(({ ...props }, ref) => {
  const TYPES_ICONS_MAP: Record<ActionType, IconType> = {
    [ActionType.ApiCall]: 'RiTerminalBoxFill',
    [ActionType.ApiCallGraphql]: 'RiTerminalWindowFill',
    [ActionType.Calculation]: 'RiCalculatorFill',
    [ActionType.Checklist]: 'RiCheckboxMultipleFill',
    [ActionType.ClinicalNote]: 'RiHealthBookFill',
    [ActionType.Form]: 'RiSurveyFill',
    [ActionType.Message]: 'RiMessage3Fill',
    [ActionType.Plugin]: 'RiUploadCloudFill',
    [ActionType.PushToEmr]: 'RiUploadCloudFill'
  };

  const { icon_url, actionType, size = 'sm' } = props;

  const icon = TYPES_ICONS_MAP[actionType as keyof typeof TYPES_ICONS_MAP] ?? 'RiQuestionFill';

  const color = icon === 'RiQuestionFill' ? 'fill-slate-400' : 'fill-slate-800';

  const SIZES = {
    sm: { icon: 'w-4 h-4', container: 'w-6 h-6' },
    lg: { icon: 'w-6 h-6', container: 'w-9 h-9' },
    xl: { icon: 'w-8 h-8', container: 'w-12 h-12' }
  } as const;

  const iconSize = SIZES[size].icon;
  const containersize = SIZES[size].container;

  if (actionType === ActionType.Plugin && icon_url) {
    return <img className={`${containersize} rounded-md border border-slate-200`} src={icon_url} />;
  }

  return (
    <div
      ref={ref}
      className={`${containersize} flex bg-slate-100 rounded-md border border-slate-200 items-center justify-center`}
    >
      <Icon icon={icon} className={cn(`${iconSize}`, color)} />
    </div>
  );
});
ActionIcon.displayName = 'ActionIcon';

export { ActionIcon };
