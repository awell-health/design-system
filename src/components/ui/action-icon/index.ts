export { ActionIcon } from './action-icon';
