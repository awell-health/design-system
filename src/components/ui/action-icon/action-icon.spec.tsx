import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';

import { ActionIcon, ActionType } from './action-icon';

describe('ActionIcon', () => {
  it('match snapshot', () => {
    const result = render(<ActionIcon />);

    expect(result).toMatchSnapshot();
  });

  it('renders plugin icon with url', () => {
    const iconUrl = 'ICON_URL';
    const { getByRole } = render(<ActionIcon actionType={ActionType.Plugin} icon_url={iconUrl} />);
    const img = getByRole('img');

    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', iconUrl);
  });
});
