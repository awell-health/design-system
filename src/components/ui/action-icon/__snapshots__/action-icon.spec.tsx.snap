// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionIcon > match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-6 h-6 flex bg-slate-100 rounded-md border border-slate-200 items-center justify-center"
      >
        <svg
          class="remixicon w-4 h-4 fill-slate-400"
          fill="default"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM11 15V17H13V15H11ZM13 13.3551C14.4457 12.9248 15.5 11.5855 15.5 10C15.5 8.067 13.933 6.5 12 6.5C10.302 6.5 8.88637 7.70919 8.56731 9.31346L10.5288 9.70577C10.6656 9.01823 11.2723 8.5 12 8.5C12.8284 8.5 13.5 9.17157 13.5 10C13.5 10.8284 12.8284 11.5 12 11.5C11.4477 11.5 11 11.9477 11 12.5V14H13V13.3551Z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-6 h-6 flex bg-slate-100 rounded-md border border-slate-200 items-center justify-center"
    >
      <svg
        class="remixicon w-4 h-4 fill-slate-400"
        fill="default"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM11 15V17H13V15H11ZM13 13.3551C14.4457 12.9248 15.5 11.5855 15.5 10C15.5 8.067 13.933 6.5 12 6.5C10.302 6.5 8.88637 7.70919 8.56731 9.31346L10.5288 9.70577C10.6656 9.01823 11.2723 8.5 12 8.5C12.8284 8.5 13.5 9.17157 13.5 10C13.5 10.8284 12.8284 11.5 12 11.5C11.4477 11.5 11 11.9477 11 12.5V14H13V13.3551Z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
